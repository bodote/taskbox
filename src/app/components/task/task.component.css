.list-item {
  font-size: 14px;
  line-height: 20px;
  display: flex;
  flex-wrap: wrap;
  height: 3rem;
  width: 100%;
  background: white;
  transition: all ease-out 150ms;

  border: 0px;
  border-bottom: 0.5px;
  border-color: lightblue;
  border-style: solid;
}
