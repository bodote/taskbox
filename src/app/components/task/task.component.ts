import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Task } from '../../models/task.model';

@Component({
  selector: 'app-task',
  templateUrl: './task.component.html',
  styleUrls: ['./task.component.css'],
})
export class TaskComponent {
  @Input() task: Task; // consists of id,title,state

  // tslint:disable-next-line: no-output-on-prefix
  @Output()
  onPinTaskC = new EventEmitter<Event>();

  // tslint:disable-next-line: no-output-on-prefix
  @Output()
  onArchiveTaskC = new EventEmitter<Event>();

  /**
   * Component method to trigger the onPin event
   * @param id string
   */
  onPin(id: any) {
    // called with task.id when click on star icon exept when already archived
    this.onPinTaskC.emit(id);
  }
  /**
   * Component method to trigger the onArchive event
   * @param id string
   */
  onArchive(id: any) {
    this.onArchiveTaskC.emit(id);
  }
}
