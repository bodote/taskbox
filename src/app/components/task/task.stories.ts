import { moduleMetadata, Story, Meta } from '@storybook/angular';

import { CommonModule } from '@angular/common';

import { action } from '@storybook/addon-actions';

import { TaskComponent } from './task.component';

export default {
  component: TaskComponent,
  decorators: [
    moduleMetadata({
      declarations: [TaskComponent],
      imports: [CommonModule],
    }),
  ],
  excludeStories: /.*Data$/,
  title: 'Task',
} as Meta;

export const actionsData = {
  onPinTaskMockAction: action('onPinTaskMockEvent'),
  onArchiveTaskMockAction: action('onArchiveTaskMockEvent'),
};

const Template: Story<TaskComponent> = (args) => ({
  props: {
    ...args,
    onPinTaskC: actionsData.onPinTaskMockAction,
    onArchiveTaskC: actionsData.onArchiveTaskMockAction,
  },
});

export const Default = Template.bind({});
Default.args = {
  task: {
    id: '1',
    title: 'Test Task',
    state: 'TASK_INBOX',
    updatedAt: new Date(2021, 0, 1, 9, 0),
  },
};

export const Pinned = Template.bind({});
Pinned.args = {
  task: {
    ...Default.args.task,
    state: 'TASK_PINNED',
  },
};

export const Archived = Template.bind({});
Archived.args = {
  task: {
    ...Default.args.task,
    state: 'TASK_ARCHIVED',
  },
};
