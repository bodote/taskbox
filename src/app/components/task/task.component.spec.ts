import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TaskComponent } from './task.component';
const puppeteer = require('puppeteer');
const { toMatchImageSnapshot } = require('jest-image-snapshot');

describe('TaskComponent', () => {
  let component: TaskComponent;
  let fixture: ComponentFixture<TaskComponent>;
  let browser;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [TaskComponent],
    }).compileComponents();
  });

  beforeEach(async () => {
    fixture = TestBed.createComponent(TaskComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should render ', () => {
    expect(fixture).toMatchSnapshot(); // This is the snapshot validation
    (expect(fixture) as any).toMatchImageSnapshot();
  });
  it('works with puppeteer', async () => {
    expect.extend({ toMatchImageSnapshot });
    browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto('http://localhost:6006/iframe.html?id=task--default&args=&viewMode=story');
    const image = await page.screenshot();
    (expect(image) as any).toMatchImageSnapshot();

    await browser.close();
  });
});
