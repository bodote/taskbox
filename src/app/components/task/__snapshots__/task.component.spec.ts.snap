// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskComponent should render  1`] = `
<app-task
  __ngContext__={[Function LRootView]}
  onArchiveTaskC={[Function EventEmitter_]}
  onPinTaskC={[Function EventEmitter_]}
>
  <div
    class="list-item"
  >
    <label
      class="checkbox"
    >
      <input
        disabled="true"
        name="checked"
        type="checkbox"
      />
      <span
        class="checkbox-custom"
      />
    </label>
    <div
      class="title"
    >
      <input
        placeholder="Input title"
        readonly="true"
        style="text-overflow: ellipsis;"
        type="text"
      />
    </div>
    <div
      class="actions"
    >
      <a>
        <span
          class="icon-star"
        />
      </a>
      
    </div>
  </div>
</app-task>
`;
